pkg/api/v1alpha1/nutanixdatacenterconfig.go

```diff
@@ -44,3 +50,13 @@ func GetNutanixDatacenterConfig(fileName string) (*NutanixDatacenterConfig, erro
 	}
 	return &clusterConfig, nil
 }
+
+// SetDefaults sets default values for the NutanixDatacenterConfig object.
+func (c *NutanixDatacenterConfig) SetDefaults() {
+	if c.Spec.CredentialRef == nil {
+		c.Spec.CredentialRef = &Ref{
+			Kind: constants.SecretKind,
+			Name: constants.NutanixCredentialsName,
```

Comment:
In the design, didn't we decide we are not going to default the secrets and make it required for the users to provide it?

Reply:
We decided that we are not going to have a defaulting webhook and creating new datacenterconfigs will be rejected if they don't have credentialRef explicitly set. For the existing ones, they need to be reconciled to add a default.